deps/rabbitmq_prometheus/src/collectors/prometheus_rabbitmq_alarm_metrics_collector.erl

```diff
@@ -0,0 +1,80 @@
+%% This Source Code Form is subject to the terms of the Mozilla Public
+%% License, v. 2.0. If a copy of the MPL was not distributed with this
+%% file, You can obtain one at https://mozilla.org/MPL/2.0/.
+%%
+%% Copyright (c) 2007-2021 VMware, Inc. or its affiliates.  All rights reserved.
+%%
+-module(prometheus_rabbitmq_alarm_metrics_collector).
+
+-export([register/0, deregister_cleanup/1, collect_mf/2]).
+
+-import(prometheus_model_helpers, [create_mf/4, untyped_metric/1]).
+
+-include_lib("prometheus/include/prometheus.hrl").
+
+-behaviour(prometheus_collector).
+
+-define(METRIC_NAME_PREFIX, "rabbitmq_alarms_").
+
+%%====================================================================
+%% Collector API
+%%====================================================================
+
+register() ->
+    ok = prometheus_registry:register_collector(?MODULE).
+
+deregister_cleanup(_) ->
+    ok.
+
+-spec collect_mf(_Registry, Callback) -> ok
+              when _Registry :: prometheus_registry:registry(),
+                   Callback :: prometheus_collector:callback().
+collect_mf(_Registry, Callback) ->
+    try
+        case rabbit_alarm:get_local_alarms(500) %% TODO: figure out timeout
+        of
+            Alarms when is_list(Alarms) ->
+                ActiveAlarms =
+                    lists:foldl(fun ({{resource_limit, disk, _}, _}, Acc) ->
+                                        maps:put(disk_limit, 1, Acc);
+                                    ({{resource_limit, memory, _}, _}, Acc) ->
+                                        maps:put(memory_limit, 1, Acc);
+                                    ({file_descriptor_limit, _}, Acc) ->
+                                        maps:put(file_descriptor_limit, 1, Acc)
+                                end,
+                                #{},
+                                Alarms),
+
+                Callback(create_mf(?METRIC_NAME(<<"file_descriptor_limit">>),
```

Comment:
"file_descriptors_used" would sound slightly more specific to me

Reply:
Or maybe specifically here "file_descriptors_limit" makes more sense. "_used" may suggest to some that it is the number of descriptors used, since booleans are represented as integers.